src/utils.zig

```diff
@@ -1929,6 +1929,12 @@ pub const BinaryReader = struct {
 		defer self.remaining = self.remaining[len + 1 ..];
 		return self.remaining[0..len :delimiter];
 	}
+
+	pub fn readSlice(self: *BinaryReader, length: usize) error{OutOfBounds, IntOutOfBounds}![]const u8 {
+		if(self.remaining.len < length) return error.OutOfBounds;
+		defer self.remaining = self.remaining[length..];
+		return self.remaining[0..length];
+	}
```

Comment:
Didn't we just add this exact function in the other PR?
Please rebase.

Reply:
Yes, we did, I was developing those two at the same time, so to avoid waiting included it here too. Since i just copy-pasted it git diff wont figure it out by itself.